alacritty.yml

```diff
@@ -613,7 +615,11 @@
   #- { key: V,                           mode: Vi, action: ToggleNormalSelection   }
   #- { key: V,      mods: Shift,         mode: Vi, action: ToggleLineSelection     }
   #- { key: V,      mods: Control,       mode: Vi, action: ToggleBlockSelection    }
+  #
+  # select the "word" under the cursor. See semantic_escape_chars to change boundary detection
```

Comment:
I don't think it makes sense to document the bindings, if anything we should document the actions.

Reply:
I'm not sure what you mean. I  _was_ documenting what `ToggleSemanticSelection` does.

Are you suggesting that there should be a separate location where actions are documented outside of the array of key bindings?

In the README the docs say _You can find the default configuration file with documentation for all available fields_ 

This gives the impression all things found in `alacritty.yml` would be documented, but the actions are not. Some are clear enough. Others are not. For example, it's not clear what "FirstOccupied" does.

Reply:
> Are you suggesting that there should be a separate location where actions are documented outside of the array of key bindings?

There already is. All existing actions are documented, you're just adding documentation about an action to a binding, which doesn't make much sense.

If we wanted to document the action, we should document it where the action is documented, not the binding.

Reply:
Where is ToggleSemanticSelection documented? 

I just searched the project to double check if I missed something:

https://github.com/alacritty/alacritty/search?q=ToggleSemanticSelection&unscoped_q=ToggleSemanticSelection

I see that `alacritty.yml` documents that it exists, but it doesn't explain what it does.

Are you referring to the comments in the source code for each action?
https://github.com/alacritty/alacritty/blob/142f84efb955a9fa59f4205ec3a6db3964c5f433/alacritty/src/config/bindings.rs#L82

The comment for `ToggleSemanticSelection` says  just "/// Toggle semantic vi selection."
That's really just repeating the words in the variable name. A more helpful comment would explain how "semantic" is implemented or provide a link for more information about the semantic matching system. That's the kind of explanation I was trying to provide. 

For example, I thought maybe it was going to match from one paren to another paren or from one brace to another brace. It wasn't clear from the name that was going to match "words".

Reply:
https://github.com/alacritty/alacritty/blob/master/alacritty.yml#L534

> I see that alacritty.yml documents that it exists, but it doesn't explain what it does.

Correct, but that's what I'm saying. If it should be documented, the action should be documented not the binding.

The binding doesn't do anything, it just ties a key to an action. You can have an action bound an arbitrary amount of times, but it still always does the same thing. So what should be documented is the action, not the binding.

> For example, I thought maybe it was going to match from one paren to another paren or from one brace to another brace. It wasn't clear from the name that was going to match "words".

It doesn't match words. It matches semantically, based on the semantic escape characters.

Reply:
I'm 100% in favor of documenting the action not the binding. The question remains, _how_ exactly to document the action?

I would propose adding them below the item (possibly with an additional newline as shown) like the following:

```
# - Open
#   Attempt to open the element under the cursor, like a URL. See also url.launcher.
#
# - ToggleSemanticSelection
#   Select the "word" under the cursor. See `semantic_escape_chars` to change boundary detection.
# - ...
```

FWIW, if this were rendered markdown it would look like this:

- Open
  Attempt to open the element under the cursor, like a URL. See also url.launcher.

- ToggleSemanticSelection
  Select the "word" under the cursor. See `semantic_escape_chars` to change boundary detection.
- ...

Reply:
I can also get behind documenting the actions and not the bindings, and @nixpulvis proposal for how to format them seems fine with me. They are in a YAML comment, for formatting them as valid Markdown does invite future automation or reformatting to further surface the docs. 

Looks like someone could take a pass through the source code to gather all the code comments describing the actions and turn those into docs.

https://github.com/alacritty/alacritty/blob/142f84efb955a9fa59f4205ec3a6db3964c5f433/alacritty/src/config/bindings.rs#L82

Reply:
I personally prefer not having spaces between the different actions, since that would waste a lot of space and break the list apart. I think something like this is better:

```
#   - Up
#       Move the vi mode cursor up
#   - Down
#       Move the vi mode cursor down
```

Personally I don't think there's much value in taking code comments into user-facing documentation. These two have completely different purposes and should not just copy one another.

Reply:
My updated commit provides docs in this format.